core-bundle/src/DependencyInjection/Attribute/AsPage.php

```diff
@@ -20,7 +20,10 @@
 #[\Attribute(\Attribute::TARGET_CLASS | \Attribute::IS_REPEATABLE)]
 class AsPage
 {
-    public function __construct(public ?string $type = null, public ?string $path = null, public array $requirements = [], public array $options = [], public array $defaults = [], public array $methods = [], string $locale = null, string $format = null, public bool $contentComposition = true, public ?string $urlSuffix = null)
+    /**
+     * @param string|bool|null $path
+     */
+    public function __construct(public ?string $type = null, public $path = null, public array $requirements = [], public array $options = [], public array $defaults = [], public array $methods = [], string $locale = null, string $format = null, public bool $contentComposition = true, public ?string $urlSuffix = null)
```

Comment:
This change seems wrong. The path should be either a string or `null`. Why did you allow `bool` as well?

Reply:
Because we can now do

```
/**
 *  @Page(path=false)
 */
```

in order to disable routing for a page.

Reply:
What does `path=null` do then?

Reply:
Oh wait, @aschempp changed it for the wrong property.

Reply:
If no `path` is defined for a Page Controller, then the behavior will be the default one, i.e. it will only use the alias for the path of the route. That's the same as before.

Reply:
It should be like this IMHO:

```php
if (null === $path) {
    if (\in_array($type, self::DISABLE_ROUTING, true)) {
        $path = '';
        $options['compiler_class'] = UnroutablePageRouteCompiler::class;
    } else {
        $path = '/'.($pageModel->alias ?: $pageModel->id).'{!parameters}';
        $defaults['parameters'] = '';
        $requirements['parameters'] = $pageModel->requireItem ? '/.+' : '(/.+?)?';
    }
}
```

Reply:
But that would not cover `@Page(path=false)`.

Reply:
Yes, I don't want to use `path=false` at all.

Reply:
How else do you want to define pages that are not routable?

Reply:
With `page=null` and the code outlined above?

Reply:
That would not allow you to disable routing for your page controller. If you set `@Page` or `@Page(path=null)` (which is equivalent) then the default behavior of only using the `alias` for the path applies - and routing will be enabled.

Reply:
because the whole point of this PR is to allow `false` as a meaning of disabling routing 😂

Reply:
@leofeyer it looks like you don't yes know how page routes work. A path can either be `null` to just be the alias, it can be relative to append the alias, or it can be absolute to override the page alias. Using `false` now adds the "disable routing" feature. We cannot use `null` for that.

Reply:
FYI I also just figured out why we have a difference between `null` and `''` (empty string). If the path is `null`, we generate the default behaviour of allowing any parameter after the alias. If the path is an empty string, the path will be exactly the alias, and no optional parameters.